editor/filesystem_dock.cpp

```diff
@@ -3634,6 +3620,47 @@ MenuButton *FileSystemDock::_create_file_menu_button() {
 	return button;
 }
 
+bool FileSystemDock::_can_dock_horizontal() const {
+	return true;
+}
+
+void FileSystemDock::_set_dock_horizontal(bool p_enable) {
+	if (button_dock_placement->is_visible() == p_enable) {
+		return;
+	}
+
+	if (p_enable) {
+		set_meta("_dock_display_mode", get_display_mode());
+		set_meta("_dock_file_display_mode", get_file_list_display_mode());
+
+		FileSystemDock::DisplayMode new_display_mode = FileSystemDock::DisplayMode(int(get_meta("_bottom_display_mode", int(FileSystemDock::DISPLAY_MODE_HSPLIT))));
+		FileSystemDock::FileListDisplayMode new_file_display_mode = FileSystemDock::FileListDisplayMode(int(get_meta("_bottom_file_display_mode", int(FileSystemDock::FILE_LIST_DISPLAY_THUMBNAILS))));
+
+		set_display_mode(new_display_mode);
+		set_file_list_display_mode(new_file_display_mode);
+		set_custom_minimum_size(Size2(0, 200) * EDSCALE);
+	} else {
+		set_meta("_bottom_display_mode", get_display_mode());
+		set_meta("_bottom_file_display_mode", get_file_list_display_mode());
+
+		FileSystemDock::DisplayMode new_display_mode = FileSystemDock::DISPLAY_MODE_TREE_ONLY;
+		FileSystemDock::FileListDisplayMode new_file_display_mode = FileSystemDock::FILE_LIST_DISPLAY_LIST;
+
+		if (has_meta("_dock_display_mode")) {
+			new_display_mode = FileSystemDock::DisplayMode(int(get_meta("_dock_display_mode")));
+		}
+		if (has_meta("_dock_file_display_mode")) {
+			new_file_display_mode = FileSystemDock::FileListDisplayMode(int(get_meta("_dock_file_display_mode")));
+		}
```

Comment:
This is what I meant by "Same below" in https://github.com/godotengine/godot/pull/86765#discussion_r1462403615

Reply:
oh oops, I missed that, sorry